### ISSUE TITLE
Android 15 edge-to-edge enforcement

### ISSUE THREAD

**Issue by ArcherEmiya05** [firebase/FirebaseUI-Android#2177]
Welcome to FirebaseUI and thanks for submitting an issue!

Please take a look at [open issues](https://github.com/firebase/FirebaseUI-Android/issues?q=is%3Aopen+is%3Aissue), as well as [resolved issues](https://github.com/firebase/FirebaseUI-Android/issues?q=is%3Aissue+is%3Aclosed), to see if your issue is either already being addressed, or has been solved by someone else.

If not, please feel free to fill in the following info so we can help faster!

### Step 1: Are you in the right place?
Yes

### Step 2: Describe your environment

  * Android device: AVD/Emulator
  * Android OS version: 15
  * Google Play Services version: N/A
  * Firebase/Play Services SDK version: N/A
  * FirebaseUI version: 8.0.2
  
### Step 3: Describe the problem:
UI is broken in Android 15 due to the new edge-to-edge enforcement.

#### Steps to reproduce:

  1. Use Firebase Android UI Auth
  2. Use Phone Auth provider
  3. Run the project
  
#### Observed Results:

 
![Screenshot_1733432497](https://github.com/user-attachments/assets/82dcce7d-fdd4-4215-ada0-d1d1b5438157)

  
#### Expected Results:

Adapt edge-to-edge or fix the UI to make it look like the way it used to on older Android versions/
  
#### Relevant Code:

A temporary fix to this is to set the below style attribute and apply the theme using `AuthUI.SignInIntentBuilder.setTheme`
  ```
   <!-- Firebase UI theme -->
    <style name="Theme.App.FirebaseAuthUI">

        <!-- For some reason, Firebase UI use this as textColorLink -->
        <item name="colorPrimary">@color/mdc_colorAccent</item>

        <!-- Use by Toolbar title color -->
        <!-- Use by Outlined text field color (unfocused) -->
        <!-- Use by OTP dashes color -->
        <!-- Use by Button disabled color -->
        <item name="colorOnSurface">@color/mdc_colorPrimary_White</item>

        <!-- TextInputLayout and button shape -->
        <item name="shapeAppearanceSmallComponent">@style/ShapeAppearance.MDC.SmallComponent</item> <!-- TODO FirebaseUI still uses MDC 2, remove this once they migrated to MDC 3 -->
        <item name="shapeAppearanceCornerExtraSmall">@style/ShapeAppearance.MDC.SmallComponent</item>

        <item name="android:windowOptOutEdgeToEdgeEnforcement" tools:targetApi="35">true</item> <!-- TODO FirebaseUI does not yet support edge-to-edge enforcement of Android 15, remove this once they adapted -->

    </style>
  ```

**Comment by thatfiredev** [comment]
@ArcherEmiya05 thanks for reporting!

We're working on this in #2187 and we'll include it as part of the next release.
In the meantime your workaround should do the trick. 👍

**Comment by ArcherEmiya05** [comment]
Hi, is there any plan to release a new version with the patch related to edge to edge enforcement since `windowOptOutEdgeToEdgeEnforcement` will no longer work when app runs on Android 16 and above.

**Comment by mnk98** [comment]
In Android 16, windowOptOutEdgeToEdgeEnforcement does not work. This is blocking us from updating our app to Android 16, which we urgently need to do. The issue has been present for a long time, but it still hasn’t been addressed. Please prioritize fixing this in Firebase Auth UI.

**Comment by ArcherEmiya05** [comment]
@thatfiredev any update on this?

**Comment by ArcherEmiya05** [comment]
> Hey everyone, We have just released [v9.1.1](https://github.com/firebase/FirebaseUI-Android/releases/tag/9.1.1) of the library which should address this issue. :)

Nope, the fix is only for Android 15 and above and doesn't include backward compatibility.



Here is how it looks like in Android 6.0

<img width="768" height="1280" alt="Image" src="https://github.com/user-attachments/assets/19da5a71-c980-4d55-b222-7ce23d5c71b4" />


I assume it didn't include calling `enableEdgeToEdge()` to support edge-to-edge on older version.

**Comment by mnk98** [comment]
Yes, that new version is not compatible with the older version of the OS.

**Comment by vegeta2102** [comment]
@russellwheatley @demolaf 
Base on this feedback, it's still not fixed yet. 
Please do not close this issue !
https://github.com/firebase/FirebaseUI-Android/issues/2177#issuecomment-3455125159
